File: app/src/test/java/com/omar/retromp3recorder/app/usecase/ShareUCImplTest.java
package com.omar.retromp3recorder.app.usecase;

import com.omar.retromp3recorder.app.share.SharingModule;

import org.junit.Test;

import io.reactivex.Completable;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

public class ShareUCImplTest {
    @Test
    public void test_UC_Executed(){
        SharingModule sharingModule = mock(SharingModule.class);
        when(sharingModule.share()).thenReturn(Completable.complete());
        ShareUCImpl shareUC = new ShareUCImpl(sharingModule);

        //When
        shareUC.execute().subscribe();

        //Then
        verify(sharingModule).share();
    }

}

File: app/src/test/java/com/omar/retromp3recorder/app/common/usecase/ShareUCImplTest.kt
package com.omar.retromp3recorder.app.common.usecase

import com.nhaarman.mockitokotlin2.whenever
import com.omar.retromp3recorder.app.share.ShareUC
import com.omar.retromp3recorder.app.share.Sharer
import io.reactivex.Completable
import org.junit.Test
import org.mockito.Mockito

class ShareUCImplTest {
    @Test
    fun `assert happy flow`() {
        val sharingModule = Mockito.mock(Sharer::class.java)
        whenever(sharingModule.share()).thenReturn(Completable.complete())
        val shareUC = ShareUC(sharingModule)

        //When
        shareUC.execute().subscribe()

        //Then
        Mockito.verify(sharingModule).share()
    }
}